src/providers/ldap/sdap_async_connection.c

```diff
@@ -1596,15 +1598,21 @@ static void sdap_cli_connect_done(struct tevent_req *subreq)
                                              struct sdap_cli_connect_state);
     const char *sasl_mech;
     int ret;
+    bool retry_same_server;
 
     talloc_zfree(state->sh);
     ret = sdap_connect_recv(subreq, state, &state->sh);
     talloc_zfree(subreq);
     if (ret) {
+        if (ret == ERR_TLS_FAILURE) {
+            retry_same_server = true;
```

Comment:
Lets add a debug message here. Something like "performing retry due to ..."

Reply:
- mistype `du`
 - I wouldn't use SSSDBG_CRIT_FAILURE here. I would propose SSSDBG_OP_FAILURE
 - I think `interrupted` -> `interruption` (but maybe better to consult with native speaker)

Reply:
I'll update the PR with the first two points corrected. As for the last one let's consult with @justin-stephenson. Justin, what's the correct way of writing the debug message?
1. "Performing retry due to TLS handshake interrupted"
2. "Performing retry due to TLS handshake interruption"

Reply:
By the way, I'm going to move this debug line to `data_provider_fo.c` file, where the retry is triggered.

Reply:
> By the way, I'm going to move this debug line to `data_provider_fo.c` file, where the retry is triggered.

Why? Information about retry reason is lost there.

Reply:
I'm changing it because each time that ldap_install_tls() fails the log will appear. The issue is that there's a maximum number of retrials (1), and the second time that the call fails the log will be printed when in reality there isn't any retrial. So I'm moving the log after the conditions to retry are processed. The conditions are `retry_same_server == true && state->svc->current_resolved_attempts <= MAX_CURRENT_RESOLVED_ATTEMPTS`.

Reply:
But at this point you don't know the reason behind `retry_same_server == true`.

I don't mind addition of a new message in FO code, but please a debug message about setting retry_same_server=true due to our assumption TLS handshake was interruped via internal WD.

Reply:
The second is correct:

"Performing retry due to TLS handshake interruption"

Reply:
I've reformulated the phrase in `sdap_async_connection.c` and I've added a new debug line in `data_provider.c`. Alexey can you take a look at it?